amqprs/src/api/connection.rs

```diff
@@ -365,6 +416,109 @@ impl OpenConnectionArguments {
     }
 }
 
+#[cfg(feature = "urispec")]
+impl TryFrom<&str> for OpenConnectionArguments {
+    type Error = Error;
+
+    /// Create a new OpenConnectionArguments from a URI &str. Mostly follows the [AMQP URI spec](https://www.rabbitmq.com/uri-spec.html). See below for exceptions.
+    ///
+    /// If the URI is invalid, a UriError error is returned.
+    ///
+    /// If no port is specified, the default port of 5672 is used (as per the spec).
+    /// If no host is speecified or is zero-length, a UriError error is returned. Note that this is different from the spec, which allows for EmptyHost. This is because the host is used to create a TCP connection, and an empty host is invalid.
+    ///
+    fn try_from(uri: &str) -> Result<Self> {
+        let pu = URIReference::try_from(uri)?;
+
+        // Check scheme
+        if pu.scheme().is_none()
+            || (pu.scheme().unwrap().as_str() != "amqp" && pu.scheme().unwrap().as_str() != "amqps")
+        {
+            return Err(Error::UriError(String::from("Invalid URI scheme")));
+        }
+
+
+        // Set the default port depending on the scheme. The unwrap should be safe due to the checks above. 
+        // Will panic if any invalid scheme is not rejected before this point
+        let default_port: &u16 = match pu.scheme().unwrap().as_str() {
```

Comment:
I think no need to use reference for u16, the reference is size of 64-bit in 64-bit machine.
just use `u16` directly, the copy is very low cost

Reply:
Good to know, I was thinking it would be less expensive to use reference. I'll switch it.